mock-kubernetes-client/client/client.go

```diff
@@ -543,6 +551,30 @@ func (client *MockClient) DeleteAllOf(context ctx.Context, object ctrlClient.Obj
 	return fmt.Errorf("Not implemented")
 }
 
+// MockStuckTermination sets a flag determining whether an object should get stuck in terminating when it is deleted.
+func (client *MockClient) MockStuckTermination(object ctrlClient.Object, terminating bool) error {
```

Comment:
With the current implementation it might be hard but I think it would be useful to set the `status.Phase` also to terminating (if we want to use this field some day).

Reply:
I considered that, but figured it was more trouble than it's worth right now because the operator doesn't check it.